CatalogServiceAPI/Providers/CatalogServiceProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿using CatalogServiceAPI.Client;
+using CatalogServiceAPI.Models.Requests;
+using CatalogServiceAPI.Models.StepsModels;
+using CatalogServiceAPI.Utils;
+using Core.Enums;
+
+namespace CatalogServiceAPI.Providers
+{
+    public class CatalogServiceProvider
+    {
+        private readonly CatalogServiceClient _catalogServiceClient;
+        private readonly ProductGenerator _productGenerator;
+
+        public CatalogServiceProvider(CatalogServiceClient client,
+           ProductGenerator generator)
+        {
+            _catalogServiceClient = client;
+            _productGenerator = generator;
+        }
+
+        public async Task CreateActiveProductWithSomePrice(CreateProductRequest request, ProductStatus status, double price)
+        {
+            await _catalogServiceClient.CreateProduct(request);
+            await _catalogServiceClient.UpdateProductPrice(request.Article, price);
+            await _catalogServiceClient.SetProductStatus(request.Article, (int)status);
+        }
+
+        public async Task CreateActiveProductsWithSomePrice(List<CreateProductRequest> productRequests, List<ProductModel> products, double price)
+        {
+            for(var i = 0; i < productRequests.Count; i++) 
+            {
+                await CreateActiveProductWithSomePrice(productRequests[i], products[i].ProductStatus, price);
+            }
+        }
```

Comment:
If these functions can create products with different statuses, you should rename these methods

Reply:
Fixed them.